rpc/internal/perf_service.go

```diff
@@ -228,21 +215,28 @@ func (srv *perfService) CloseMetrics(ctx context.Context, end *MetricsSeriesEnd)
 		return nil, newRPCError(codes.Internal, errors.Wrapf(err, "closing perf result record '%s'", record.ID))
 	}
 
-	resp.Success = true
-	return resp, nil
-}
+	hasEventData, err := srv.addFTDCRollupsJob(ctx, record.ID, record.Artifacts)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating FTDC rollups job")
+	}
 
-func (srv *perfService) addArtifacts(ctx context.Context, record *model.PerformanceResult, artifacts []*ArtifactInfo) error {
-	for _, a := range artifacts {
-		record.Artifacts = append(record.Artifacts, *a.Export())
+	// Only enqueue a new update time series job for mainline results if
+	// there are user-submitted rollups AND no event data (FTDC artifact).
+	// In the latter case, the FTDC rollups job will take care of
+	// enqueueing the update time series job.
+	if record.Info.Mainline && len(record.Rollups.Stats) > 0 && !hasEventData {
+		processingJob := units.NewUpdateTimeSeriesJob(record.CreateUnanalyzedSeries())
+		if err = amboy.EnqueueUniqueJob(ctx, srv.env.GetRemoteQueue(), processingJob); err != nil {
+			return nil, newRPCError(codes.Internal, errors.Wrapf(err, "creating signal processing job for perf result '%s'", record.ID))
+		}
 	}
```

Comment:
Why is this moved from the CreateMetrics function to CloseMetrics function?

Reply:
To ensure we only send it once per result and don't send duplicate notifications to SPS.